policy/release/pre_build_script_task/pre_build_script_task.rego

```diff
@@ -0,0 +1,88 @@
+#
+# METADATA
+# title: Pre-build-script task checks
+# description: >-
+#   This package verifies that the pre-build-script tasks in the
+#   attestation are executed in a controlled environment
+#
+package pre_build_script_task
+
+import rego.v1
+
+import data.lib
+import data.lib.sbom
+import data.lib.tekton
+
+# METADATA
+# title: Script runner image is listed in the task results
+# description: >-
+#   Verify that the image used to run the pre-build script task is
+#   listed in the task output SCRIPT_RUNNER_IMAGE_REFERENCE
+# custom:
+#   short_name: pre_build_script_task_runner_image_in_results
+#   failure_msg: >-
+#     The runner image used for the pre-Build-Script task '%s' is not
+#     listed in the task results
+#   solution: >-
+#     Make sure the image used to run the pre-build task is referenced
+#     in the 'SCRIPT_RUNNER_IMAGE_REFERENCE' task result.
+#   collections:
+#   - redhat
+#
+deny contains result if {
+	some attestation in lib.pipelinerun_attestations
+	some task in tekton.pre_build_tasks(attestation)
+	not tekton.task_result(task, _pre_build_run_script_runner_image_result)
```

Comment:
Not a big deal, and I'm not suggesting to change it, but I notice the actual check is doing a bit less than what the description in the metadata says. I.e. this would still pass if any old garbage was in the task result value, right?